eva-server/src/main/java/uk/ac/ebi/eva/server/ws/ga4gh/GA4GHBeaconWSServerV2.java

```diff
@@ -0,0 +1,325 @@
+                                               @RequestParam(value = "startMax", required = false) Long startMax,
+                                               @RequestParam(value = "end", required = false) Long end,
+                                               @RequestParam(value = "endMin", required = false) Long endMin,
+                                               @RequestParam(value = "endMax", required = false) Long endMax,
+                                               @RequestParam(value = "referenceBases") String referenceBases,
+                                               @RequestParam(value = "alternateBases", required = false)
+                                                       String alternateBases,
+                                               @RequestParam(value = "variantType", required = false) String variantType,
+                                               @RequestParam(value = "assemblyId") String assemblyId,
+                                               @RequestParam(value = "datasetIds", required = false) List<String> studies,
+                                               @RequestParam(value = "includeDatasetResponses", required = false)
+                                                       String includeDatasetResponses,
+                                               HttpServletResponse response)
+            throws IOException, AnnotationMetadataNotFoundException {
+
+        initializeQuery();
+        BeaconAlleleRequestBody request = new BeaconAlleleRequestBody(chromosome, start, startMin, startMax, end,
+                endMin, endMax, referenceBases, alternateBases, variantType, assemblyId, studies,
+                includeDatasetResponses);
+
+        if (assemblyId.equalsIgnoreCase("grch37")) {
+            MultiMongoDbFactory.setDatabaseNameForCurrentThread(DBAdaptorConnector.getDBName("hsapiens_grch37"));
+        } else if (assemblyId.equalsIgnoreCase("grch38")) {
+            MultiMongoDbFactory.setDatabaseNameForCurrentThread(DBAdaptorConnector.getDBName("hsapiens_grch38"));
+
+        } else {
+            return new GA4GHBeaconQueryResponseV2(GA4GHBeaconResponseV2.id_val, GA4GHBeaconResponseV2.apiVersion_val,
+                    null, request, new BeaconError(HttpServletResponse.SC_BAD_REQUEST,
+                    "Please enter a valid assemblyId"), null);
+        }
+
+        String errorMessage = checkErrorHelper(request);
+
+        if (errorMessage != null) {
+            response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+            return new GA4GHBeaconQueryResponseV2(GA4GHBeaconResponseV2.id_val,
+                    GA4GHBeaconResponseV2.apiVersion_val, null, request,
+                    new BeaconError(HttpServletResponse.SC_BAD_REQUEST, errorMessage), null);
+        }
+
+        VariantType variantType1;
+        try {
+            variantType1 = VariantType.valueOf(variantType);
+        } catch (Exception e) {
+            variantType1 = null;
```

Comment:
That's true. 
I will check this condition in the `checkErrorHelper` function  so that even if an invalid  variantType comes, it will be returned as invalid initially itself :+1:

Reply:
Fixed :+1: